src/store/UsersReducer.js

```diff
@@ -0,0 +1,43 @@
+
+const INITIAL_STATE = {
+  current: {
+    email: "",
+  },
+  available: [ "alice@gmail.com", "bob@gmail.com", "sammy@gmail.com"],
+};
+
+const usersReducer = (state = INITIAL_STATE, action) => {
+  const {
+    current,
+    available,
+  } = state;
+  switch (action.type) {
+    case 'ADD_USER':
+     
+      const addedUser = action.email;
+
+      // And put user in users.available
+      available.push(addedUser);
+
+      // Finally, update the redux state
+      return {
+        ...state, 
+          current: {email: action.email}
+      }
```

Comment:
Podemos simplificar e tornar mais explicito o que estamos a alterar no reducer
```suggestion
    case 'ADD_USER':
      return {
          available: [ ... available, action.email],
          current: {email: action.email}
      }
```